File: freshman/src/main/java/com/mredrock/cyxbs/freshman/mvp/presenter/BasePresenter.java
package com.mredrock.cyxbs.freshman.mvp.presenter;


import com.mredrock.cyxbs.freshman.mvp.contract.BaseContract;

public class BasePresenter<V extends BaseContract.ISomethingView> implements IBasePresenter<V> {
    private V view;

    //关联View
    @Override
    public void attachView(V view) {
        this.view = view;
    }

    //解除View关联
    @Override
    public void detachView() {
        view = null;
    }

    public V getView() {
        return view;
    }
}

File: freshman/src/main/java/com/mredrock/cyxbs/freshman/utils/kt/BasePresenter.kt
package com.mredrock.cyxbs.freshman.utils.kt

import com.mredrock.cyxbs.freshman.mvp.presenter.IBasePresenter

abstract class BasePresenter<V, M> : IBasePresenter<V> {
    protected var mvpView: V? = null
        private set

    protected val model by lazy { createModel() }

    override fun attachView(mvpView: V) {
        this.mvpView = mvpView
    }

    override fun detachView() {
        mvpView = null
    }

    abstract fun createModel(): M

}